### PR TITLE
Fix the issue gosec-G306 G307

### LINKED ISSUE #32: Builds: Static and Error checking of Go code
**Enhancement Description**
In order to facilitate coding best practices, this adapter's code should have static analysis and error checking run as part of its build process.

**Chore Description**
Using the meshery-istio adapter's Actions workflow as an example, enable [Static and Error checks](https://github.com/layer5io/meshery-istio/blob/master/.github/workflows/ci.yml#L26-L51) in this repository's [CI.yml workflow](../blob/master/.github/workflows/ci.yml).

### LINKED ISSUE #44: The CI Security check catches the defer function issue with G306 G307
**Description**
A brief description of the issue.

* The `defer` function in the adapter's source code got the gosec(Golang Security Checker)-G307. To check the [available rule list](https://github.com/securego/gosec#available-rules) 
  

**Expected Behavior**
A brief description of what you expected to happen.

* All the test scripts will be pass

**Screenshots**
Add screenshots, if applicable, to help explain your problem.

**Enviroment:**
 - OS: [e.g. Ubuntu]
 - Browser: [e.g. Chrome, Safari]
 - Version: [e.g. 22]
 - Device: [e.g. laptop, iPhone 8]

---
[Optional] **To Reproduce**

[Optional] **Additional Context**

### PR DESCRIPTION
**Pull request by Aisuko**
Signed-off-by: Aisuko <urakiny@gmail.com>

FIxes #44
Fixes #32

### CODE REVIEW

File: linkerd/linkerd.go [hunk 1]
```diff
@@ -290,7 +290,7 @@ func (iClient *Client) executeInstall(ctx context.Context, arReq *meshes.ApplyRu
 	var tmpKubeConfigFileLoc = path.Join(os.TempDir(), fmt.Sprintf("kubeconfig_%d", time.Now().UnixNano()))
 	// -L <namespace> --context <context name> --kubeconfig <file path>
 	// logrus.Debugf("about to write kubeconfig to file: %s", iClient.kubeconfig)
-	if err := ioutil.WriteFile(tmpKubeConfigFileLoc, iClient.kubeconfig, 0700); err != nil {
+	if err := ioutil.WriteFile(tmpKubeConfigFileLoc, iClient.kubeconfig, 0600); err != nil {
```
Comment (Aisuko) [inline linkerd/linkerd.go:293]: I have a consider on `gosec-G306`, Should we ignore the create file permission check? The kubeconfig file under the 0600 only can be r/d, but under 0700 can be r/d/e. I'm not sure the what's the permission of kubeconfig in this adapter completely

File: linkerd/install.go [hunk 2]
```diff
@@ -118,7 +118,15 @@ func (iClient *Client) downloadFile(urlToDownload, localFile string) error {
 		logrus.Error(err)
 		return err
 	}
-	defer dFile.Close()
+
+	defer dFile.Close() // #nosec
```
Comment (Aisuko) [inline linkerd/install.go:122]: I am willing to accept the tradeoff of more readable and easily understandable code for needing to obsessively review code to ensure that the file is closed in all cases, and that’s the approach I recommend in code reviews I give to others.

File: linkerd/linkerd.go (hunk 1 above)
Comment (leecalcote) [reply linkerd/linkerd.go:293]: Good of you to ask. I can’t think of a reason that kubeconfig would need to be executable. Unless I’m missing something, we should be ok to move to 600.

File: linkerd/install.go (hunk 2 above)
Comment (nveenjain) [reply linkerd/install.go:122]: @Aisuko, can we create a callFuncAndLogError function that takes function as a parameter and execute the function and if error occurred, just log it, we can then always defer this function by passing the function that needs to be deferred as a parameter...

Comment (Aisuko) [reply linkerd/install.go:122]: @nveenjain Good point. I know what's your mean, the ` func` as a parameter. I'd like to try the solution we discussed. Maybe after the Github action Issue

Comment (leecalcote) [reply linkerd/install.go:122]: Gents, I'm withholding opinion here. Given your discussion, shall we merge this and circle back with a GitHub Action?

**Comment by Aisuko** [comment]
Hi @leecalcote The PR includes the `ci` changed. Because I found some issues with KinD(https://github.com/engineerd/setup-kind/pull/20). So, I believe our `ci.yaml`  also could include the scripts which I added to `KinD`.

And there also have an issue with the ci `Security check` and it looks like `go get` command's issue. I'm working on slove it.

Screen Shot 2020-04-08 at 11 07 39 AM

**Comment by Aisuko** [comment]
@leecalcote Please take a look. I believe we can merge the PR with all CI passed.

**Pull request merged**